src/FloatingButton/FloatingButton.tsx

```diff
@@ -0,0 +1,198 @@
+
+  offsetX: PropTypes.number,
+  offsetY: PropTypes.number,
+  positionX: PropTypes.oneOf(["left", "center", "right"]),
+  positionY: PropTypes.oneOf(["bottom", "top"]),
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  animate: true,
+  label: "+",
+  offsetX: 30,
+  offsetY: 30,
+  positionX: "right",
+  positionY: "bottom",
+};
+
+export default class FloatingButton extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      active: false,
+      display: false,
+    };
+  }
+
+  horizontalStyle() {
+    const { offsetX, positionX } = this.props;
+    if (positionX === "center") return {};
+    return positionX === "left"
+      ? {
+          left: `${offsetX}px`,
+        }
+      : {
+          // default to right if a bad positionX input provided
+          right: `${offsetX}px`,
+        };
+  }
+
+  verticalStyle() {
+    const { offsetY, positionY } = this.props;
+    return positionY === "top"
+      ? {
+          top: `${offsetY}px`,
+        }
+      : {
+          // default to bottom orientation
+          bottom: `${offsetY}px`,
+        };
+  }
+
+  mainButtonHandler = () => {
+    const { additionalButtons, animate, onClick } = this.props;
+    const { active } = this.state;
+    if (additionalButtons && additionalButtons.length) {
+      // main button activates additional buttons
+      if (active) {
+        // stop displaying when transition to inactive state complete
+        this.setState({ active: false, display: true });
+        setTimeout(() => this.setState({ display: false }), animate ? 200 : 0);
+      } else {
+        this.setState({ active: true, display: true });
+      }
+    }
+    if (onClick && !active) {
+      onClick();
+    }
+  };
+
+  render() {
+    const {
+      additionalButtons,
+      animate,
+      bgColor,
+      className,
+      closeLabel,
+      label,
+      positionX,
+      positionY,
+      size,
+    } = this.props;
+    const { active, display } = this.state;
+
+    return (
+      <FlexBox
+        className={classnames(
+          cssClass.CONTAINER,
+          cssClass.propStyle(positionX),
+          cssClass.propStyle(positionY),
+          className,
+        )}
+        style={{
+          ...this.horizontalStyle(),
+          ...this.verticalStyle(),
+        }}
+      >
+        <div className={classnames(cssClass.SUB_ELEMENT, className)}>
+          <Button
+            className={cssClass.BUTTON}
+            onClick={this.mainButtonHandler}
+            value={
+              active
+                ? closeLabel || (
+                    <div>
+                      <span className="fa fa-times" /> Close
+                    </div>
+                  )
+                : label
+            }
+            size={size || "regular"}
+            style={{
+              borderRadius: "500px",
+              ...(bgColor
```

Comment:
Rather than allowing users to specify any background color, we should define an enum for colors and internally map those to Dewey colors.

```js
export const Color = {
  BLUE: "blue",
  GREEN: "green",
  ...
};

const cssClass = {
  colorCssClass: (color: string) => `FloatingButton--color--${color}`,
  ...
};

const propTypes = {
  color: PropTypes.oneOf(Object.values(Color)),
  ...
};
```

```less
.FloatingButton--color--blue {
  background-color: @primary_blue;
}

.FloatingButton--color--green {
  background-color: @alert_green;
}
```

Reply:
It's my intention to do exactly this except in the Button component. I talked with Neil about it. It's a little hacky the way I have it but I'm hoping it's ok to do it this way temporarily.

Reply:
Got it! Makes perfect sense to push this into the `<Button>` long term.

In the meantime though, I'm still hesitant to expose a generic color prop instead of a pre-specified enum color prop. Especially because transitioning from the generic color prop to the enum will be a breaking change. And having a generic prop allows consumers of the component to break from Design-approved colors.

Even when we do add color enum logic to the `<Button>`, the `<FloatingButton>` will need to expose the same color prop and pass it through to the `<Button>`. So might as well set up that target interface, even if we take the halfway approach of only supporting the color prop in `<FloatingButton>` and not `<Button>`